File: solutions/89-wahid/age-of-villagers/age-of-villagers/factory/nationfactory.cs
using age_of_villagers.nation;
using System;
using System.Collections.Generic;
using System.Text;

namespace age_of_villagers.factory
{
    class nationfactory
    {
        string name;
        public nation.Ination Getnation(string nationtype)
        {
            if (nationtype == "bangladeshi farmers")
            {
                return new bangladeshifarmers();
            }
            else if (nationtype == "arab bedouin")
            {
                return new arabbedouin();
            }
            else if (nationtype == "egytian kings")
            {
                return new egyptiankings();
            }
            else if (nationtype == "inuit hunters")
            {
                return new inuithunters();
            }
            throw new Exception();

        }
    }
}

File: solutions/58-sakib-2/src/Factory/NationFactory.java
package Factory;

import Nations.*;

public class NationFactory {
    public INation getNation(String nation){
        if (nation=="Bangladeshi Farmers")return new BangladeshiFarmers();
        else if (nation=="Arab Bedouin")return new ArabBedouin();
        else if (nation=="Egyptian Kings")return new EgyptianKings();
        else if (nation=="Inuit Hunters")return new InuitHunters();
        else return null;
    }
}
